runtime-feature-detection.md

```diff
@@ -0,0 +1,156 @@
+# Runtime Feature Detection on Graviton
+
+Different generations of Graviton support different features. For example, Graviton3 supports SVE but Graviton2 does
+not. Graviton4 supports SVE2 and SVE. For some applications it may be adventageous to implement performance critical
+kernels to make use of the highest performing feature set available which may not be known until runtime. For this, the
+best practice is to consult HWCAPS. This guide covers the procedure for feature detection on Linux.
+
+On Linux, the standard C library implements `getauxval` to read from the auxiliary vector. Reading the `AT_HWCAP` and
+`AT_HWCAP2` values will return the hardware capability bitmaps which are filtered by the kernel to include only features
+which the kernel also supports. The library function is defined in `sys/auxv.h` and the constants for masking the bitmap
+are defined in `asm/hwcap.h`. Below is a simple C program to print all of the hardware capabilities and print an
```

Comment:
I suggest we upload this code as a .c file, this way users can more directly retrieve it with curl if they want to run a quick test.

More important than the code, I would like to see a table of the features which are and are not supported on the hardware, much like you posted in the PR.   This way customers have direct reference to what they can expect, and if they aren't getting it they should investigate the kernel or re-verify their logic.

Reply:
I added this table and posted the code in a standalone file.